config/hocon-mp/src/main/java/io/helidon/config/hocon/mp/HoconMpConfigIncluder.java

```diff
@@ -0,0 +1,134 @@
+
+    private ConfigParseOptions parseOptions;
+    private String relativeUrl;
+    private Path relativePath;
+    private Charset charset;
+
+    HoconMpConfigIncluder() {
+    }
+
+    @Override
+    public ConfigIncluder withFallback(ConfigIncluder fallback) {
+        return this;
+    }
+
+    @Override
+    public ConfigObject include(ConfigIncludeContext context, String what) {
+        new Exception().printStackTrace();
+        LOGGER.log(TRACE, String.format("Received request to include resource %s, %s",
+                what, context.parseOptions().getOriginDescription()));
+
+        return relativeUrl != null ? parseHoconFromUrl(what) : parseHoconFromPath(what);
+    }
+
+    private ConfigObject parseHoconFromUrl(String includeName) {
+        String includePath = relativeUrl + patchName(includeName);
+        URL includeUrl;
+        try {
+            includeUrl = new URL(includePath);
+            System.out.println("includeURL: " + includeUrl);
+        } catch (MalformedURLException e) {
+            LOGGER.log(WARNING, String.format("Unable to create include Url for: %s with error: %s",
+                    includePath, e.getMessage()));
+            return ConfigFactory.empty().root();
+        }
+        try (InputStreamReader readable = new InputStreamReader(includeUrl.openConnection().getInputStream(), charset)) {
+            Config typesafeConfig = ConfigFactory.parseReader(readable, parseOptions);
+            return typesafeConfig.root();
+        } catch (IOException e) {
+            throw new ConfigParserException("Failed to read from include source: " + includeUrl, e);
+        }
+    }
+
+    private ConfigObject parseHoconFromPath(String includeName) {
+        Path path = relativePath.resolve(includeName);
+        if (Files.exists(path) && Files.isReadable(path) && !Files.isDirectory(path)) {
+            System.out.println("Path: " + path);
+            try (BufferedReader reader = Files.newBufferedReader(path, charset)) {
+                Config typesafeConfig = ConfigFactory.parseReader(reader, parseOptions);
+                return typesafeConfig.root();
+            } catch (IOException e) {
+                throw new ConfigException("Failed to read from include source: " + path.toAbsolutePath(), e);
+            }
+        } else {
+            return ConfigFactory.empty().root();
+        }
+    }
+
+    void charset(Charset charset) {
+        this.charset = charset;
+    }
+
+    void parseOptions(ConfigParseOptions parseOptions) {
+        this.parseOptions = parseOptions;
+    }
+
+    void relativeUrl(String relativeUrl) {
+        this.relativeUrl = relativeUrl;
+    }
+
+    void relativePath(Path relativePath) {
+        this.relativePath = relativePath;
+    }
+
+    /**
+     * Adds default Hocon extension if not present.
+     *
+     * @param what file name
+     * @return file name with extension
+     */
+    public static String patchName(String what) {
```

Comment:
This method should not be public (class is not public)

Reply:
Strange, I thought it was not public. But updating this.

Reply:
Done, changed to private